sound/soc/sof/pcm.c

```diff
@@ -288,9 +288,41 @@ static int sof_pcm_hw_free(struct snd_soc_component *component,
 	struct snd_sof_pcm *spcm;
 	int ret, err = 0;
 
-	/* nothing to do for BE */
-	if (rtd->dai_link->no_pcm)
+	if (rtd->dai_link->no_pcm) {
```

Comment:
Interesting, this new IPC message is only needed and only sent in the "no PCM" case, right? Presumably, otherwise the same clean up is triggered in the firmware by an already existing and used IPC? Is a corresponding firmware change already available?

Reply:
Yes, which is the backend DAI. I forgot to mention the FW counterpart:

https://github.com/thesofproject/sof/pull/4262